File: ecomerceBacked/src/main/java/com/ecomerce/ecomerceBacked/service/ProductService.java
package com.ecomerce.ecomerceBacked.service;

import com.ecomerce.ecomerceBacked.model.Product;
import com.ecomerce.ecomerceBacked.repository.ProductRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProductService  {
    private final ProductRepository productRepository;

    public ProductService(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }
    //To get all products
    public List<Product> getAllProducts(){
        return productRepository.findAll();
    }
    public Product getProductById(Long id) {
        return productRepository.findById(id).orElse(null);
    }
    // create product
    public  Product createProduct(Product product){
        return productRepository.save(product);
    }//update products
    public Product updateProduct(Long id,Product product){
        product.setId(id);
        return productRepository.save(product);
    }
    public List<Product> getProductsByCategory(Long categoryId) {
        return productRepository.findByCategoryId(categoryId);
    }
    //delete product
    public void deleteProduct(Long id){
        productRepository.deleteById(id);
    }

}

File: ecommerce-frontend/src/services/productService.js
import API from "./api";

export const getAllProducts = () => API.get("/products").then((r) => r.data);
export const getProductById = (id) => API.get(`/products/${id}`).then((r) => r.data);
export const getProductsByCategory = (categoryId) =>
  API.get(`/products/category/${categoryId}`).then((r) => r.data);
export const createProduct = (data) => API.post("/products", data).then((r) => r.data);
export const updateProduct = (id, data) => API.put(`/products/${id}`, data).then((r) => r.data);
export const deleteProduct = (id) => API.delete(`/products/${id}`).then((r) => r.data);
